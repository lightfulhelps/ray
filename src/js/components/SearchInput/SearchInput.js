// @flow
import * as React from 'react';
import classNames from 'classnames';
import { FormInput, Button } from '../../';

type Props = {
  className?: string,
  onChange: (SyntheticKeyboardEvent<HTMLInputElement>) => void,
  onSearch: (
    SyntheticKeyboardEvent<HTMLInputElement> | SyntheticMouseEvent<HTMLButtonElement>
  ) => void,
  placeholder?: string,
  value: string,
};

const SearchInput = ({
  className,
  onChange,
  onSearch,
  placeholder = 'Search...',
  value,
  ...other
}: Props) => {
  const classes = classNames(className, 'search-input input-group shadow');
  const handleKeyPress = (e: SyntheticKeyboardEvent<HTMLInputElement>) => {
    if (e.key === 'Enter') {
      onSearch(e);
    }
  };

  return (
    <div {...other} className={classes}>
      <FormInput
        data-test-id="search-input"
        onChange={onChange}
        onKeyPress={handleKeyPress}
        placeholder={placeholder}
        value={value}
      />
      <div className="input-group-append">
        <Button
          data-test-id="search-input-button"
          icon="search"
          isOutline
          onClick={onSearch}
          theme="gray-600"
        />
      </div>
    </div>
  );
};

export default SearchInput;
