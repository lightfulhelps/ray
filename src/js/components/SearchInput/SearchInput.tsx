import * as React from 'react';
import classNames from 'classnames';
import { FormInput, Button } from '../..';

interface Props extends React.HTMLAttributes<HTMLInputElement> {
  className?: string;
  onSearch: (arg0: React.KeyboardEvent<HTMLInputElement> | React.MouseEvent) => void;
  placeholder?: string;
  value: string;
}

const SearchInput: React.FC<Props> = ({
  className,
  onChange,
  onSearch,
  placeholder = 'Search...',
  value,
  ...other
}: Props) => {
  const classes = classNames(className, 'search-input input-group shadow');
  const handleKeyPress = (e: React.KeyboardEvent<HTMLInputElement>) => {
    if (e.key === 'Enter') {
      onSearch(e);
    }
  };

  return (
    <div {...other} className={classes}>
      <FormInput
        data-test-id="search-input"
        onChange={onChange}
        onKeyPress={handleKeyPress}
        placeholder={placeholder}
        type="text"
        value={value}
      />
      <div className="input-group-append">
        <Button
          data-test-id="search-input-button"
          icon="search"
          isOutline
          onClick={onSearch}
          theme="gray-600"
        />
      </div>
    </div>
  );
};

export default SearchInput;
