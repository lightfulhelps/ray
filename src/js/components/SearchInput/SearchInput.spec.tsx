import React from 'react';
import { shallow } from 'enzyme';
import merge from 'lodash/merge';
import SearchInput from './SearchInput';

const setup = (overrides = {}) => {
  const props = merge(
    {
      onChange: jest.fn(),
      onSearch: jest.fn(),
    },
    overrides
  );
  const wrapper = shallow(<SearchInput {...props} />);

  return { wrapper, props };
};

describe('<SearchInput />', () => {
  it('should render', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('should handle className', () => {
    const { wrapper } = setup();

    wrapper.setProps({ className: 'custom' });

    expect(wrapper.hasClass('search-input')).toBe(true);
    expect(wrapper.hasClass('custom')).toBe(true);
  });

  it('should handle onChange', () => {
    const { wrapper, props } = setup();

    wrapper.find('[data-test-id="search-input"]').simulate('change', { target: { value: 'Test' } });

    expect(props.onChange).toHaveBeenCalledWith({ target: { value: 'Test' } });
  });

  describe('onSearch', () => {
    it('should call onSearch on button click', () => {
      const { wrapper, props } = setup();

      wrapper.find('[data-test-id="search-input-button"]').simulate('click');

      expect(props.onSearch).toHaveBeenCalledTimes(1);
    });

    it('should call onSearch with event on Enter keypress', () => {
      const { wrapper, props } = setup();

      wrapper.find('[data-test-id="search-input"]').simulate('keyPress', { key: 'a' });

      expect(props.onSearch).toHaveBeenCalledTimes(0);

      wrapper.find('[data-test-id="search-input"]').simulate('keyPress', { key: 'Enter' });

      expect(props.onSearch).toHaveBeenCalledTimes(1);
      expect(props.onSearch).toHaveBeenCalledWith({ key: 'Enter' });
    });
  });

  it('should handle placeholder', () => {
    const { wrapper } = setup();

    expect(wrapper.find('[data-test-id="search-input"]').prop('placeholder')).toEqual('Search...');

    wrapper.setProps({ placeholder: 'Search here...' });

    expect(wrapper.find('[data-test-id="search-input"]').prop('placeholder')).toEqual(
      'Search here...'
    );
  });

  it('should handle value', () => {
    const { wrapper } = setup({ value: 'Test' });

    expect(wrapper.find('[data-test-id="search-input"]').prop('value')).toEqual('Test');
  });

  it('should pass through other props', () => {
    const { wrapper } = setup({ tabIndex: 1, id: 'test' });

    expect(wrapper.prop('tabIndex')).toEqual(1);
    expect(wrapper.prop('id')).toEqual('test');
  });
});
