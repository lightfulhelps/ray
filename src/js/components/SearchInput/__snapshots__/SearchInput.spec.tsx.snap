// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput /> should render 1`] = `
<div
  className="search-input input-group shadow"
>
  <FormInput
    data-test-id="search-input"
    onChange={[MockFunction]}
    onKeyPress={[Function]}
    placeholder="Search..."
    type="text"
  />
  <Button
    className="input-group-append"
    data-test-id="search-input-button"
    icon="search"
    isOutline={true}
    onClick={[MockFunction]}
    theme="gray-600"
  />
</div>
`;
